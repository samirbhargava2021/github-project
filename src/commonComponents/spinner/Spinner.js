import React from "react";
import './Spinner.scss';
import { CMS } from '../../cms/CMS';
const Spinner = () => {
  return (
    <div className="spinner-position">
      <div className="spinner"/>
      <p>{CMS.loading}</p>
    </div>
  );
};

export default Spinner;
