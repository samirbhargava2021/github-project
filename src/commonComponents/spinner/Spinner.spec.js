import { shallow } from "enzyme";
import { render } from "@testing-library/react";
import Spinner from "./Spinner";

describe("Spinner Component", () => {
  it("should render spinner Component", () => {
    const component = shallow(<Spinner />);
    expect(component.getElements()).toMatchSnapshot();
  });
  it("should have className spinner to load the spinner with proper styling", () => {
    const { container } = render(<Spinner />);
    expect(container.firstChild.firstChild.classList.contains('spinner')).toBe(true)
  });
  it("should have text 'Loading.....' in the component", () => {
    const { getByText } = render(<Spinner />);
    expect(getByText("Loading.....")).toBeDefined();
  });
  it("should have text 'Loading.....' in the component", () => {
    const { getByText } = render(<Spinner />);
    expect(getByText("Loading.....")).toBeDefined();
  });
});
