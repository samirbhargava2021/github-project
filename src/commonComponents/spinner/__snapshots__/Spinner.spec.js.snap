// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component should render spinner Component 1`] = `
Array [
  <div
    className="spinner-position"
  >
    <div
      className="spinner"
    />
    <p>
      Loading.....
    </p>
  </div>,
]
`;
